src/app/modules/main-module/components/main/main.component.ts

```diff
@@ -103,15 +125,17 @@ export class MainComponent implements OnInit {
   }
 
   private setStatus() {
-    for (let i = 0; i < this.letterStatus.length; i++) {
-      this.wordStatus[i] = this.letterStatus[i].status;
-    }
+    let pos = 0;
+    this.letterStatus.forEach((value) => {
```

Comment:
no te hace falta poner un contador el forEach adminte un index  this.letterStatus.forEach((value, index), cuando tengáis dudas leer la documentación de estas cosas